lib/model/Product.dart

```diff
@@ -461,4 +479,23 @@ class Product extends JsonObject {
     }
     return result;
   }
+
+  /// Returns all the potential improvements given the quality of the data
+  ///
+  /// For apps with contribution mode.
+  /// A typical use-case is to alert the end-users that they can improve
+  /// the quality of the OFF data by taking pictures (or something like that),
+  /// when displaying a [Product].
+  Set<ProductImprovement> getProductImprovements() {
+    final Set<ProductImprovement> result = {};
+    if (statesTags != null) {
```

Comment:
Minor but I'd avoid nested if structures like this, if possible. You could change it by adding something like
```
if (statestags == null) {
    return result;
}
```

Reply:
Thank you @peterwvj for your comment.
I agree with you, I'm a big fan myself of `return`, `break` or `continue` statements instead of `else`s (nested or not).
The thing is that I also had in mind #193, that will impact the very same method.
When I implement #193 (next week), I'll take your remark into account if still relevant.